src/pykeen/nn/modules.py

```diff
@@ -2549,17 +2550,77 @@ def forward(
         )
 
 
+Clamp = tuple[float | None, float] | tuple[float, float | None]
+
+
+class ClampedInteraction(Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]):
+    """An adapter to clamp scores to a minimum or maximum value.
+
+    .. warning::
+        The used :func:`torch.clamp` function has zero gradient for scores below the minimum of above the maximum value.
+        Thus, it aggravates gradient-based optimization.
+    """
+
+    clamp_score: Clamp | None
+
+    # TODO: this seems to cause cyclic imports during sphinx
+    # @update_docstring_with_resolver_keys(ResolverKey(name="base", resolver="pykeen.nn.modules.interaction_resolver"))
+    def __init__(
+        self,
+        clamp_score: Clamp | float | None = None,
+        base: HintOrType[
+            Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]
+        ] = DistMultInteraction,
```

Comment:
not sure if it makes sense to have a default for this one

Reply:
Probably not, since the `interaction_resolver` already has a default. 🙂

Reply:
even with a default for that I think that this kind of interaction "middleware" should only be applied to something that's explicit, so maybe we can abuse the type annotation to tell people they really shouldn't make a clamp interaction without knowing the other one they want to use